packages/astro/src/core/preview/index.ts

```diff
@@ -30,7 +30,10 @@ export default async function preview(config: AstroConfig, { logging }: PreviewO
 	const trailingSlash = config.devOptions.trailingSlash
 	/** Base request URL. */
 	let baseURL = new URL(config.buildOptions.site || '/', defaultOrigin);
-
+	const staticFileServer = sirv(fileURLToPath(config.dist), {
```

Comment:
Fix: we shouldn't create a new instance on every request. This should speed up performance considerably if a site requests many assets per page.